driver/src/orderbook/streamed/balance.rs

```diff
@@ -98,24 +98,34 @@ impl Balance {
     }
 
     pub fn sell(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.sell_buy_internal(amount, batch_id, TradeType::Sell)
+        self.sell_buy_internal(amount.into(), batch_id, TradeType::Sell)
     }
 
     pub fn buy(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.sell_buy_internal(amount, batch_id, TradeType::Buy)
+        self.sell_buy_internal(amount.into(), batch_id, TradeType::Buy)
     }
 
     pub fn revert_sell(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.revert_sell_buy_internal(amount, batch_id, TradeType::Sell)
+        self.revert_sell_buy_internal(amount.into(), batch_id, TradeType::Sell)
     }
 
     pub fn revert_buy(&mut self, amount: u128, batch_id: BatchId) -> Result<()> {
-        self.revert_sell_buy_internal(amount, batch_id, TradeType::Buy)
+        self.revert_sell_buy_internal(amount.into(), batch_id, TradeType::Buy)
+    }
+
+    pub fn solution_submission(&mut self, fee: U256, batch_id: BatchId) -> Result<()> {
+        // We are reusing the `Proceeds` machinery here because the extra balance from a solution
+        // submission behaves the same way.
+        self.sell_buy_internal(fee, batch_id, TradeType::Buy)
+    }
+
+    pub fn revert_solution_submission(&mut self, fee: U256, batch_id: BatchId) -> Result<()> {
+        self.revert_sell_buy_internal(fee, batch_id, TradeType::Buy)
```

Comment:
should this be of type sell?

Reply:
I imagine its a buy as you are reverting receiving some amount. That's how I understood it at least.

Reply:
It should be a buy. The same way when reverting a buy we also use TradeType::Buy because it is going to subtract from the increase field.